extensions/wikibase/src/org/openrefine/wikibase/schema/WbVariableExpr.java

```diff
@@ -90,12 +92,20 @@ public void setColumnName(String columnName) {
      * Evaluates the expression in a given context, returning
      * 
      * @throws QAWarningException
+     * @throws SpecialValueNoValueException
+     * @throws SpecialValueSomeValueException
      */
     @Override
     public T evaluate(ExpressionContext ctxt)
-            throws SkipSchemaExpressionException, QAWarningException {
+            throws SkipSchemaExpressionException, QAWarningException, SpecialValueNoValueException, SpecialValueSomeValueException {
         Cell cell = ctxt.getCellByName(columnName);
         if (cell != null) {
+            if (cell.toString() == "#NOVALUE#") {
+                throw new SpecialValueNoValueException();
+            } else if (cell.toString() == "#SOMEVALUE#") {
+                throw new SpecialValueSomeValueException();
+            }
```

Comment:
It's probably worth storing those special strings in some `public static final String` constant declared somewhere, reused thorough the codebase, especially because they will be used by a lot of different classes.

Also, we should not use ` == ` to compare Strings, but the `.equals()` method. It is customary to write the constant first (so that you are sure that you are calling the `.equals()` on a non-null object) -> so `"#NOVALUE#".equals(cell.toString())`.

Reply:
> It's probably worth storing those special strings in some public static final String constant declared somewhere, reused thorough the codebase, especially because they will be used by a lot of different classes.

Maybe in the "entityvalues" definitions?

>Also, we should not use == to compare Strings, but the .equals() method. It is customary to write the constant first (so that you are sure that you are calling the .equals() on a non-null object) -> so "#NOVALUE#".equals(cell.toString()).
Yeah I have a couple of things like that which I need to cleanup.